cdap-api/src/main/java/co/cask/cdap/api/customaction/AbstractCustomAction.java

```diff
@@ -67,11 +71,13 @@ public final void initialize(CustomActionContext context) throws Exception {
    * Classes derived from {@link AbstractCustomAction} can override this method to initialize the {@link CustomAction}.
    * @throws Exception if there is any error in initializing the custom action
    */
+  @TransactionPolicy(TransactionControl.IMPLICIT)
```

Comment:
That is why we have special handling in the CustomActionExecutor: it checks whether the action extends AbstractCustomAction, and if so, it determines the transaction policy of initialize(). 

That means, for certain program types (MR, Spark, CustomAction), we have special handling in the platform, which is aware that the abstract program implement a final initialize(context) and requires initialize() from the subclass instead.

This is covered by test cases.